include/exadg/solvers_and_preconditioners/utilities/petsc_operation.h

```diff
@@ -96,9 +89,35 @@ apply_petsc_operation(VectorType &                                           dst
       dst.local_element(i) = ptr[i];
     }
 
-    ierr = VecRestoreArray(vector_dst, &ptr);
+    ierr = VecRestoreArray(petsc_vector_dst, &ptr);
     AssertThrow(ierr == 0, ExcPETScError(ierr));
   }
+}
+
+/*
+ *  This function wraps the copy of a PETSc object (sparse matrix,
+ *  preconditioner) with a dealii::LinearAlgebra::distributed::Vector,
+ *  allocating a PETSc vectors and then calling the other function
+ */
+template<typename VectorType>
+void
+apply_petsc_operation(VectorType &                                           dst,
+                      VectorType const &                                     src,
+                      std::function<void(PETScWrappers::VectorBase &,
+                                         PETScWrappers::VectorBase const &)> petsc_operation,
+                      MPI_Comm const &                                       petsc_mpi_communicator)
```

Comment:
Is this function still needed?

Reply:
We use it for the `Operator::vmult_matrix_based` (used for benchmarking) where we do not want to hold a `Vec` object of PETSc, and also for the case we set up a CG solver with an AMG preconditioner (again because I did not want to pollute the interface with more objects than just PETSc things). Also, that part is probably less performance-critical than the `PreconditionerBoomerAMG` class because the run time will typically be higher, so I opted for the smaller code size there.